src/core/webview/ClineProvider.ts

```diff
@@ -68,42 +68,37 @@ export class ClineProvider extends EventEmitter<ClineProviderEvents> implements
 	public static readonly tabPanelId = "roo-cline.TabPanelProvider"
 	private static activeInstances: Set<ClineProvider> = new Set()
 	private disposables: vscode.Disposable[] = []
-	// not private, so it can be accessed from webviewMessageHandler
-	view?: vscode.WebviewView | vscode.WebviewPanel
-	// not private, so it can be accessed from webviewMessageHandler
-	// callers could update to get viewLaunched() getter function
-	isViewLaunched = false
+	private view?: vscode.WebviewView | vscode.WebviewPanel
 	private clineStack: Cline[] = []
-	// not private, so it can be accessed from webviewMessageHandler
-	workspaceTracker?: WorkspaceTracker
-	// not protected, so it can be accessed from webviewMessageHandler.
-	// Could modify code to use getMcpHub() instead.
-	mcpHub?: McpHub // Change from private to protected
-	// not private, so it can be accessed from webviewMessageHandler
-	latestAnnouncementId = "mar-30-2025-3-11" // update for v3.11.0 announcement
-	// not private, so it can be accessed from webviewMessageHandler
-	settingsImportedAt?: number
+	private _workspaceTracker?: WorkspaceTracker // workSpaceTracker read-only for access outside this class
```

Comment:
workspaceTracker needs to be public to allow access to initializeFilePaths() but we can make it readonly externally.

can't apply the TS readonly property because we need to write to this field in our dispose() method, so this is the next best option in terms of limiting public access.